include/ceed-impl.h

```diff
@@ -302,6 +302,16 @@ struct CeedFortranContext_private {
 };
 typedef struct CeedFortranContext_private *CeedFortranContext;
 
+struct CeedContextFieldLabel_private {
+  const char *name;
+  const char *description;
+  CeedContextFieldType type;
+  size_t size;
+  size_t offset;
+  CeedInt num_sub_labels;
+  CeedContextFieldLabel *sub_labels;
+};
```

Comment:
Do we need a tree here or could we make a flat array with the address of all the structs? Perhaps
```c
struct {
  char *host, *device;
  size_t size, offset;
} *locations;

for (i=0; i<n; i++) {
  memcpy(locations[i].host + locations[i].offset, &value_to_set, bytes_of_value);
}
```
followed by an async copy to device (if needed).

Reply:
My plan was to let the GPU backends handle a copy to the GPU if it is needed and only update the CPU side in the interface.

I don't understand why an array of offsets is better than an array of related labels?

Reply:
Probably premature optimization, but also trying to make this structure flat and simple so it doesn't get abused for more semantic operations. Your solution is fine -- we'll be able to refactor if it becomes necessary.

Reply:
Yep, good point. We should do that.